### DISCUSSION TITLE
Renovate fails to parse a `composer.json` that contains `"repositories": {"type": "gitlab", ...}` metadata

### DISCUSSION THREAD

**Discussion by Ocramius** [renovatebot/renovate#23185] [Request Help]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

`ghcr.io/renovatebot/renovate:35.24.5-slim@sha256:8250bd834fb31f3a0f0bf907f9ec5a016fd05d1d0a437255e73366111c705965`

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I have a PHP application that contains a `composer.json` similar to this:

```json
{
    "name": "my/project",
    "repositories": {
        "a-repo": {
            "type": "gitlab",
            "url": "https://gitlab.com/some/project/path.git"
        }
    },
}
```

The logic for this kind of composer repository seems to be missing:

https://github.com/renovatebot/renovate/blob/0bcd387b89f866172f43aa536b1ee954502280dc/lib/modules/manager/composer/schema.ts#L63-L170

While parsing `composer.json`, the entire `"repositories"` structure seems to be therefore discarded.

### What you tried so far.

1. installed renovate runner on a `gitlab.com` system
2. ran it against the project at hand (via `$RENOVATE_EXTRA_FLAGS`)
3. observed output / noticed incompatibility (see logs below)

Note: for now, I held back from changing the repository that renovate is scanning, and instead came here to report this first issue I found. I may be able to replace `"type": "gitlab"` in the next few days, if I'm given green light on that.

I can gladly provide some sort of reproducer as a test addition to https://github.com/renovatebot/renovate/blob/0bcd387b89f866172f43aa536b1ee954502280dc/lib/modules/manager/composer/schema.spec.ts#L94-L99, if wished.

Also, I'm unsure if there are differences in handling `"vcs"`, `"github"` and `"gitlab"` repository types :thinking: 

Related: https://github.com/renovatebot/renovate/discussions/22086

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Composer: error parsing repositories object (repository=<snip-omitted-for-customer-privacy>, baseBranch=develop, branch=update/magento-magento-cloud-metapackage-2.x)
       "err": {
         "message": "Schema error",
         "stack": "ZodError: Schema error\n    at Object.transform (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/util/schema-utils.ts:189:21)\n    at /opt/containerbase/tools/renovate/35.159.6/node_modules/zod/lib/types.js:2852:51\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at /opt/containerbase/tools/renovate/35.159.6/node_modules/zod/lib/types.js:1914:29\n    at async Promise.all (index 0)\n    at /opt/containerbase/tools/renovate/35.159.6/node_modules/zod/lib/types.js:1704:32\n    at /opt/containerbase/tools/renovate/35.159.6/node_modules/zod/lib/types.js:1704:32\n    at ZodEffects.safeParseAsync (/opt/containerbase/tools/renovate/35.159.6/node_modules/zod/lib/types.js:180:24)\n    at Object.extractPackageFile (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/modules/manager/composer/extract.ts:9:15)\n    at checkBranchDepsMatchBaseDeps (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/update/branch/auto-replace.ts:121:17)\n    at checkExistingBranch (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/update/branch/auto-replace.ts:143:9)\n    at doAutoReplace (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/update/branch/auto-replace.ts:186:12)\n    at getUpdatedPackageFiles (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/update/branch/get-updated.ts:168:19)\n    at processBranch (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/update/branch/index.ts:450:19)\n    at writeUpdates (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/process/write.ts:159:17)\n    at update (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/process/extract-update.ts:214:11)\n    at Object.renovateRepository (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/repository/index.ts:75:19)\n    at attributes.repository (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /opt/containerbase/tools/renovate/35.159.6/node_modules/renovate/lib/renovate.ts:18:22",
         "issues": {
           "type": "Invalid discriminator value. Expected 'composer' | 'vcs' | 'git' | 'path' | 'package'"
         }
       }
```

</details>

**Comment by rarkins** [comment]
Please first test against latest Renovate version, and if the problem remains then create a reproduction repo
